web/src/components/onboarding/NumericField.tsx

```diff
@@ -1,53 +1,19 @@
-import { OnboardingField } from "@/components/onboarding/OnboardingField";
+import { OnboardingField, OnboardingProps } from "@/components/onboarding/OnboardingField";
 import { OnboardingDefaults } from "@/display-defaults/onboarding/OnboardingDefaults";
-import { ProfileFields } from "@/lib/types/types";
 import { templateEval } from "@/lib/utils/helpers";
-import React, { FocusEvent, ReactElement } from "react";
+import React, { ReactElement } from "react";
+import { NumericFieldProps } from "../GenericNumericField";
 
-interface Props {
-  onValidation: (field: ProfileFields, invalid: boolean) => void;
-  invalid: boolean;
-  fieldName: ProfileFields;
-  validationText?: string;
-  maxLength: number;
-  minLength?: number;
-  disabled?: boolean;
-}
-
-export const NumericField = (props: Props): ReactElement => {
-  const onValidation = (event: FocusEvent<HTMLInputElement>): void => {
-    const userInput = event.target.value.length;
-
-    const valid = props.minLength
-      ? userInput >= props.minLength && userInput <= props.maxLength
-      : userInput === props.maxLength || userInput === 0;
-
-    props.onValidation(props.fieldName, !valid);
-  };
-
-  const handleChange = (): void => props.onValidation(props.fieldName, false);
-
-  const regex = (value: string): string => value.replace(/[^0-9]/g, "");
+interface Props extends Omit<OnboardingProps, "numericProps">, NumericFieldProps {}
 
+export const NumericField = ({ minLength, maxLength, ...props }: Props): ReactElement => {
```

Comment:
if this is specific to onboarding we should probably call it OnboardingNumericField

Reply:
but also again, now that it doesn't have any of its own logic, why does this component need to exist at all?